### PR TITLE
Add Absolute url matcher

### PR DESCRIPTION
**Pull request by victorg1991**
### Checklist

- [x] I've checked that all new and existing tests pass
- [x] I've updated the documentation if necessary
- [x] I've added an entry in the CHANGELOG to credit myself

### Description

This PR adds a matcher for the url matcher

### Motivation and Context

Sometimes it's useful to match urls by the absolute url

I've created a test to changes the changes and run all the test 100% success

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # OHHTTPStubs — CHANGELOG
 
 ## Master
+  Added absoulte url matcher.
```
Comment (Liquidsoul) [inline CHANGELOG.md]: Small typo here ✏️

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # OHHTTPStubs — CHANGELOG
 
 ## Master
+  Added absoulte url matcher.  
+  [@victorg1991](https://github.com/victorg1991
+  [#252](https://github.com/AliSoftware/OHHTTPStubs/pull/252)
```
Comment (Liquidsoul) [inline CHANGELOG.md]: The number of this PR is 254 😉

File: OHHTTPStubs/Sources/Swift/OHHTTPStubsSwift.swift [hunk 3]
```diff
@@ -158,6 +158,20 @@ public func isMethodDELETE() -> OHHTTPStubsTestBlock {
...
+ * Matcher for testing an `NSURLRequest`'s **absolute url**.
+ *
+ * e.g. the absolute url is `https://api.example.com/signin` in `https://api.example.com/signin`
+ *
+ * - Parameter scheme: The absolute url to match
+ *
+ * - Returns: a matcher (OHHTTPStubsTestBlock) that succeeds only if the request
+ *            has the given absolute url
+ */
+public func isAbsoluteUrl(_ url: String) -> OHHTTPStubsTestBlock {
```
Comment (Liquidsoul) [inline OHHTTPStubs/Sources/Swift/OHHTTPStubsSwift.swift]: I am wondering if this should not be called `isAbsoluteString` to match the `URL` member we call underneath, because there is a `URL.absoluteURL`
🤔

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,9 @@
 # OHHTTPStubs — CHANGELOG
 
 ## Master
+  Added absolute url matcher.
```
Comment (AliSoftware) [inline CHANGELOG.md]: Don't forget the `*` to create a bullet point

File: OHHTTPStubs/Sources/Swift/OHHTTPStubsSwift.swift [hunk 5]
```diff
@@ -158,6 +158,20 @@ public func isMethodDELETE() -> OHHTTPStubsTestBlock {
 }
 
 /**
+ * Matcher for testing an `NSURLRequest`'s **absolute url**.
+ *
+ * e.g. the absolute url is `https://api.example.com/signin` in `https://api.example.com/signin`
```
Comment (AliSoftware) [inline OHHTTPStubs/Sources/Swift/OHHTTPStubsSwift.swift]: Maybe use a more contrived example like with `x=1&y=2#anchor` query + anchor?

File: OHHTTPStubs/Sources/Swift/OHHTTPStubsSwift.swift [hunk 6]
```diff
@@ -158,6 +158,20 @@ public func isMethodDELETE() -> OHHTTPStubsTestBlock {
...
+ *
+ * - Parameter scheme: The absolute url to match
+ *
```
Comment (AliSoftware) [inline OHHTTPStubs/Sources/Swift/OHHTTPStubsSwift.swift]: `Parameter url:`

File: OHHTTPStubs/Sources/Swift/OHHTTPStubsSwift.swift (hunk 3 above)
Comment (AliSoftware) [inline OHHTTPStubs/Sources/Swift/OHHTTPStubsSwift.swift]: Totally. We tried to name most if not all our matchers from the name of the method we call underneath to make the relationship clear, use common vocabulary and avoid confusion so let's try to continue that way 😉

File: CHANGELOG.md (hunk 1 above)
Comment (victorg1991) [inline CHANGELOG.md]: done!

File: OHHTTPStubs/UnitTests/Test Suites/SwiftHelpersTests.swift [hunk 7]
```diff
@@ -52,6 +52,28 @@ class SwiftHelpersTests : XCTestCase {
...
   }
 
+  func testIsAbsoluteUrl() {
+    let matcher = isAbsoluteUrl("foo://foo/bar")
+
+    let urls = [
+      "foo:": false,
+      "foo://foo/bar/baz": false,
+      "foo://foo/bar": true,
+      "bar://foo/bar": false,
```
Comment (AliSoftware) [inline OHHTTPStubs/UnitTests/Test Suites/SwiftHelpersTests.swift]: Maybe add one or two more complex examples, I'm especially thinking about URLs with query parameters and anchors etc

File: OHHTTPStubs/Sources/Swift/OHHTTPStubsSwift.swift (hunk 3 above)
Comment (victorg1991) [inline OHHTTPStubs/Sources/Swift/OHHTTPStubsSwift.swift]: I think that `isAbsoluteUrl` has more meaning than the other option, but its up to you.
Do I have to change to `isAbsoluteString`? I could also do something like URL.absoluteURL.absoluteString ¯\_(ツ)_/¯  haha
What do you think?

Comment (AliSoftware) [inline OHHTTPStubs/Sources/Swift/OHHTTPStubsSwift.swift]: The problem with `isAbsoluteUrl` is also that it doesn't follow the Swift naming convention, letting us believe that the expected parameter is an `URL` and not a `String`.
Otoh just "isAbsoluteString" without context (as it's a free function, not a method on a tour) is also ambiguous… maybe `isAbsoluteURLString`? But might be too long. Tbh I honestly don't know what's best suited (especially at 0:30am 😅)

Comment (victorg1991) [inline OHHTTPStubs/Sources/Swift/OHHTTPStubsSwift.swift]: I prefer `isAbsoluteURLString` over `isAbsoluteString`, it doesn't sound too bad :)

Comment (victorg1991) [inline OHHTTPStubs/Sources/Swift/OHHTTPStubsSwift.swift]: So I rename to `isAbsoluteURLString`? @AliSoftware

Comment (AliSoftware) [inline OHHTTPStubs/Sources/Swift/OHHTTPStubsSwift.swift]: yeah seems like a good compromise 👍

**Comment by victorg1991** [comment]
Done!

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,6 +1,9 @@
 # OHHTTPStubs — CHANGELOG
 
 ## Master
+* Added absolute url matcher.  
+  [@victorg1991](https://github.com/victorg1991
```
Comment (Liquidsoul) [inline CHANGELOG.md]: Typo here, missing closing parenthesis ✏️

Comment (victorg1991) [inline CHANGELOG.md]: Dish! I'll fix tha in a minute : D

Comment (victorg1991) [inline CHANGELOG.md]: Done 👍

File: OHHTTPStubs/UnitTests/Test Suites/SwiftHelpersTests.swift [hunk 9]
```diff
@@ -52,6 +52,31 @@ class SwiftHelpersTests : XCTestCase {
...
+      "bar://foo/ba": false,
+      "foobar://": false
+    ]
+
+    for (url, result) in urls {
+      #if swift(>=3.0)
+        let req = URLRequest(url: URL(string: url)!)
+      #else
+        let req = NSURLRequest(URL: NSURL(string: url)!)
```
Comment (AliSoftware) [inline OHHTTPStubs/UnitTests/Test Suites/SwiftHelpersTests.swift]: Unrelated to this PR directly, but I see we have that pattern on multiple tests: maybe we should add a `func request(from string: String) -> URLRequest` helper function for our tests, responsible for handling Swift 2/3 compatibility, and use that in our test cases to avoid repeating those `#if Swift(>=3.0)` everywhere? (Might be a separate PR though)

Comment (Liquidsoul) [inline OHHTTPStubs/UnitTests/Test Suites/SwiftHelpersTests.swift]: Yes, I agree. But I think this should be in another PR.
